jibreen/my_fastApi_env/main.py

```diff
@@ -0,0 +1,153 @@
+    
+    if not urgent_tasks:
+        raise HTTPException(status_code=404, detail="No urgent tasks found")
+    
+    return urgent_tasks
+
+
+# Endpoint for retrieving task by ID
+@app.get("/tasks/{task_id}/", summary="Get Task by ID", description="Retrieve the details of a task by its ID.")
+def get_task_by_id(
+    task_id: UUID = Path(..., description="The ID of the task to retrieve")
+):
+    tasks = load_tasks()
+    
+    # Find the task by its ID
+    for task in tasks:
+        if task.get("task_id") == str(task_id):
+            return task
+    
+    # If task not found
+    raise HTTPException(status_code=404, detail="Task not found")
+
+
+# Your create_task_or_urgent_task endpoint
+@app.post("/tasks/", summary="Create a new task", description="Create a new task with details like title, description, due date, status, and priority.")
+def create_task_or_urgent_task(  
+    task_create: TaskCreate,
+    priority: Optional[PriorityEnum] = Query(None, description="Specify the priority if the task is urgent")
+):
+    tasks = load_tasks()
+
+    # Determine if it's an urgent task or a normal task based on the presence of priority
+    if priority:
+        task = UrgentTask(
+            task_id=uuid4(),
+            title=task_create.title,
+            description=task_create.description,
+            due_date=task_create.due_date,
+            status=task_create.status,
+            priority=priority
+        )
+    else:
+        task = Task(
+            task_id=uuid4(),
+            title=task_create.title,
+            description=task_create.description,
+            due_date=task_create.due_date,
+            status=task_create.status
+        )
+
+    # Convert task to dictionary for storage
+    task_dict = task.to_dict()
+
+    # Add new task to the list
+    tasks.append(task_dict)
+    save_tasks(tasks)
+    
+    return task_dict
+
+
+@app.delete("/tasks/{task_id}",summary="Delete a Task by ID" , description="Delete a task by its ID.")
+def delete_task_by_id(task_id: UUID = Path(..., description="The ID of the task to delete")):
+    tasks = load_tasks()
+    
+    # Check if the task exists
+    if any(task.get("task_id") == str(task_id) for task in tasks):
+        # Remove the task with the matching ID
+        tasks = [task for task in tasks if task.get("task_id") != str(task_id)]
+        save_tasks(tasks)  # Save the updated task list
+        return {"detail": "Task deleted successfully"}
+    
+    # If task not found
+    raise HTTPException(status_code=404, detail="Task not found")
+
+@app.put("/tasks/{task_id}", summary="Update a task", description="Update a task by its ID.")
+def update_task_by_id(
+    *,
+    task_id: UUID = Path(..., description="The ID of the task to update"),
+    task_update: TaskUpdate
+):
+    tasks = load_tasks()
+
+    # Find the index of the task with the given task_id
+    task_index = next((index for (index, t) in enumerate(tasks) if t.get("task_id") == str(task_id)), None)
```

Comment:
Since you are lopping over the array, you can check if the task id the same and update the fields directly instead if get the index and then update the values